packages/node/src/lib/novu.ts

```diff
@@ -34,13 +36,53 @@ export class Novu extends EventEmitter {
     super();
     this.apiKey = apiKey;
 
-    this.http = axios.create({
+    const axiosInstance = axios.create({
       baseURL: this.buildBackendUrl(config),
       headers: {
         Authorization: `ApiKey ${this.apiKey}`,
       },
     });
 
+    axiosInstance.interceptors.request.use((axiosConfig) => {
+      const idempotencyKey = axiosConfig.headers['Idempotency-Key'];
+      // that means intercepted request is retried, so don't generate new idempotency key
+      if (idempotencyKey) {
+        return axiosConfig;
+      }
+
+      axiosConfig.headers['Idempotency-Key'] = uuid();
+
+      return axiosConfig;
+    });
+
+    const retryConfig = config?.retryConfig || {};
```

Comment:
shouldn't this retry be opt-in? I mean there should be a check if retryConfig exists instead of making use of default values?

Reply:
There is 0 retries by default.